homeassistant/components/auth/__init__.py

```diff
@@ -293,24 +293,26 @@ async def _async_handle_auth_code(self, hass, data, remote_addr):
                 status_code=HTTP_BAD_REQUEST,
             )
 
-        user = self._retrieve_user(client_id, RESULT_TYPE_USER, code)
+        cred = self._retrieve_user(client_id, RESULT_TYPE_CREDENTIALS, code)
```

Comment:
Name `retrieve_user` is a bit odd if it returns a credential

Reply:
Well it can return either. It depends on the passed parameter oddly.

I think a further pull request could clean this duality up and remove the ability to every return user. So i think it's better to avoid changing this here.